erpnext/setup/install.py

```diff
@@ -29,12 +29,12 @@ def after_install():
 
 
 def check_setup_wizard_not_completed():
-	if frappe.db.get_default('desktop:home_page') == 'desktop':
-		print()
-		print("ERPNext can only be installed on a fresh site where the setup wizard is not completed")
-		print("You can reinstall this site (after saving your data) using: bench --site [sitename] reinstall")
-		print()
-		return False
+	if frappe.db.get_default('desktop:home_page') != 'setup-wizard':
+		message = """Cannot install ERPNext.
+
+ERPNext can only be installed on a fresh site where the setup wizard is not completed.
+You can reinstall this site (after saving your data) using: bench --site [sitename] reinstall"""
+		frappe.throw(message, exc=frappe.IncompatibleApp)
```

Comment:
This issue is related to the current site state and not app compatibility. Perhaps better if we don't raise `IncompatibleApp` here.

```suggestion
		frappe.throw(message)
```

Reply:
@gavindsouza The default exception is ValidationError, I think IncompatibleApp is better than that.

Reply:
`IncompatibleApp` seems okay in this case. 

Apart from that, since currently `IncompatibleApp` is used in the case of incompatible versions, we can add `IncompatibleVersion` error class for it instead?